java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthViewVector.java

```diff
@@ -1345,7 +1366,29 @@ public void copyFrom(int fromIndex, int thisIndex, ValueVector from) {
    */
   @Override
   public void copyFromSafe(int fromIndex, int thisIndex, ValueVector from) {
-    throw new UnsupportedOperationException("copyFromSafe is not supported for VariableWidthVector");
+    Preconditions.checkArgument(this.getMinorType() == from.getMinorType());
```

Comment:
We generally don't write explicit `this` outside of constructors

Reply:
Ah, right. I will clean up.